### DISCUSSION TITLE
The lock entity should validate the `code_format` and entity features

### DISCUSSION THREAD

**Discussion by jbouwh** [home-assistant/architecture#861] [Entity Models] [answered]
The lock entity does not validate the code using the regular expression that can be assigned to `_attr_code_format` when calling the service lock `open`, `lock` or `unlock`.
The same applies to `LockEntityFeature.OPEN`, which is not validated when calling service lock `open`.
I would say this should be dealt with from the `lock` component, and not by the integrations that use the `lock` component.

**Comment by frenck** [comment, marked as answer]
Validation of existing documented functionality is always fine to add.

The only thing that needs consideration is: Does it need to be soft at first (warning) or hard (raise).

Go ahead, no architectural approval for validating is needed.

../Frenck

**Reply by jbouwh** [reply]
https://github.com/home-assistant/core/pull/85842
